server/tserver/src/main/java/org/apache/accumulo/tserver/log/DfsLogger.java

```diff
@@ -443,8 +444,13 @@ public synchronized void open(String address) throws IOException {
       byte[] cryptoParams = encrypter.getDecryptionParameters();
       CryptoUtils.writeParams(cryptoParams, logFile);
 
-      encryptingLogFile = new DataOutputStream(
-          encrypter.encryptStream(new NoFlushOutputStream(logFile)));
+      // write directly to logFile if no crypto detected
+      if (encrypter instanceof NoFileEncrypter) {
+        encryptingLogFile = logFile;
+      } else {
+        encryptingLogFile = (NoFlushOutputStream) encrypter
```

Comment:
Does the pluggable crypto service have to return a `NoFlushOutputStream`?  Or does Accumulo code wrapping the crypto service create the `NoFlushOutputStream`?

Reply:
The crypto service needs the NoFlushOutputStream when writing to the WAL. Perhaps we should have separate methods for the two types of encryption, instead of just passing in the scope.

Reply:
I rewrote this line of code, to no longer cast.  Not sure why I didn't do it this in the first place...